internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -627,6 +623,15 @@ func setCondition(condition hivev1.ClusterDeploymentCondition, conditions *[]hiv
 	}
 }
 
+func clearClusterAPIErrors(cluster *hivev1.ClusterDeployment) {
+	conditions := &cluster.Status.Conditions
+	for i := len(*conditions) - 1; i >= 0; i-- {
+		if (*conditions)[i].Reason == AgentPlatformError {
```

Comment:
I guess that since we currently aggregate by Reason and not by type, you cannot use:

https://github.com/openshift/custom-resource-status/blob/420d9ecf2a00ede3326466a446300ab925ae1ddc/conditions/v1/conditions.go#L49-L57

@rollandf: maybe something to address in general when you write the conditions?

Reply:
yes, and since I need the index of the condition and not just the condition this function doesn't help so much (in addition we can get more than one error).